db/multi.go

```diff
@@ -0,0 +1,105 @@
+package db
+
+import (
+	"fmt"
+	"strings"
+)
+
+// MultiInserter makes it easy to construct a
+// `INSERT INTO table (...) VALUES ... RETURNING id;`
+// query which inserts multiple rows into the same table. It can also execute
+// the resulting query.
+type MultiInserter struct {
+	table     string
+	fields    string
+	retCol    string
+	numFields int
+	values    [][]interface{}
+}
+
+// newMultiInserter creates a new MultiInserter, checking for reasonable table
+// name and list of fields.
+func NewMultiInserter(table string, fields string, retCol string) (*MultiInserter, error) {
+	numFields := len(strings.Split(fields, ","))
+	if len(table) == 0 || len(fields) == 0 || numFields == 0 {
+		return nil, fmt.Errorf("empty table name or fields list")
+	}
+	if strings.Contains(retCol, ",") {
+		return nil, fmt.Errorf("return column must be singular: %q", retCol)
+	}
+
+	return &MultiInserter{
+		table:     table,
+		fields:    fields,
+		retCol:    retCol,
+		numFields: numFields,
+		values:    make([][]interface{}, 0),
+	}, nil
+}
+
+// add another row to be included in the query
+func (mi *MultiInserter) Add(row []interface{}) error {
+	if len(row) != mi.numFields {
+		return fmt.Errorf("field count mismatch: got %d, expected %d", len(row), mi.numFields)
```

Comment:
```suggestion
		return fmt.Errorf("field count mismatch, got %d, expected %d", len(row), mi.numFields)
```

I think `got %d, expected %d` should only be preceded by `:` if it were the contents of another error message.